src/BlockTypes/ProductCategories.php

```diff
@@ -109,7 +119,30 @@ protected function get_container_classes( $attributes = array() ) {
 			$classes[] = 'is-list';
 		}
 
-		return implode( ' ', $classes );
+		$line_height_class_and_style = StyleAttributesUtils::get_line_height_class_and_style( $attributes );
+		$text_color_class_and_style  = StyleAttributesUtils::get_text_color_class_and_style( $attributes );
+		$font_size_class_and_style   = StyleAttributesUtils::get_font_size_class_and_style( $attributes );
+
+		$line_height_class = isset( $line_height_class_and_style['class'] ) ? $line_height_class_and_style['class'] : null;
+		$text_color_class  = isset( $text_color_class_and_style['class'] ) ? $text_color_class_and_style['class'] : null;
+		$font_size_class   = isset( $font_size_class_and_style['class'] ) ? $font_size_class_and_style['class'] : null;
+
+		$classes = array_filter(
+			array( $line_height_class, $text_color_class, $font_size_class )
+		);
+
+		$line_height_style = isset( $line_height_class_and_style['style'] ) ? $line_height_class_and_style['style'] : null;
+		$text_color_style  = isset( $text_color_class_and_style['style'] ) ? $text_color_class_and_style['style'] : null;
+		$font_size_style   = isset( $font_size_class_and_style['style'] ) ? $font_size_class_and_style['style'] : null;
+
+		$styles = array_filter(
+			array( $line_height_style, $text_color_style, $font_size_style )
+		);
```

Comment:
I'm wondering if we can use `StyleAttributesUtils:get_classes_and_styles_by_attributes` here? Something like this:
```suggestion
		foreach( StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes ) as $key => $item ) {
			// We skip link_color for wrapper element.
			if ( 'link_color' === $key ) {
				continue;
			}

			if ( ! empty( $item['class'] ) ) {
				$classes[] = $item['class'];
			}

			if ( ! empty( $item['style'] ) ) {
				$styles[] = $item['style'];
			}
		}
```

Reply:
>I'm wondering if we can use StyleAttributesUtils:get_classes_and_styles_by_attributes here? Something like this:

Maybe in the future, we could add more attributes and if this happens we should update this `foreach`. I would prefer to use an immutable approach. What do you think?

Reply:
@gigitux I agree with an immutable approach. But I think we still need too much code to calculate classes/styles string to use in markup. I have an idea to refactor the util in #5277, please take a look!

Reply:
> I agree with an immutable approach. But I think we still need too much code to calculate classes/styles string to use in markup. I have an idea to refactor the util in #5277, please take a look!

Done and approved! Thanks for your great feedback!